engine/src/main/java/org/pentaho/di/base/MetaFileCacheImpl.java

```diff
@@ -39,19 +40,23 @@ public class MetaFileCacheImpl implements IMetaFileCache {
   }
 
   @Override public JobMeta getCachedJobMeta( String key ) {
-    return cacheMap.get( key ) == null ? null : ( (JobMeta) cacheMap.get( key ).getMeta() );
+    String keyNormalized = key.endsWith( ".kjb" ) ? key : key + ".kjb";
```

Comment:
Strings ".kjb" and ".ktr" should be extracted to a constant to avoid multiple instantiations.